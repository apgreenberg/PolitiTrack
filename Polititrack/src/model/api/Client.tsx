import { ApolloClient, InMemoryCache } from '@apollo/client';

import OPENSTATES_OAUTH_TOKEN from '../../types/env';

const client = new ApolloClient({
  uri: 'https://openstates.org/graphql',
  headers: {
    'X-API-KEY': `${OPENSTATES_OAUTH_TOKEN}`,
  },
  cache: new InMemoryCache(),
});

export default client;
