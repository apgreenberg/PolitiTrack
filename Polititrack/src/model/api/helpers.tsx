export enum Status {
  Loading,
  Error,
  Success,
}
