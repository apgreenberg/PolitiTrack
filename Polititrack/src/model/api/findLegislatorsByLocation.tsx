import { gql, useQuery } from '@apollo/client';
import { Status } from './helpers';

export interface LegislatorSearchResults {
  status: Status;
  errorMessage: string;
  data: LegislatorSearchNode[];
}

export interface LegislatorSearchNode {
  id: string;
  name: string;
}

export const FIND_LEGISLATORS_BY_LOCATION_QUERY = gql`
  query findLegislatorByLocation($latitude: Float!, $longitude: Float!) {
    people(latitude: $latitude, longitude: $longitude, first: 10) {
      edges {
        node {
          id
          name
        }
      }
    }
  }
`;

/**
 * Gets the user's legislators based on the location of their device
 * @param requestParams - the request parameters (latitude, longitude)
 * @returns - the status of the request (loading, error or success) and the user's legislators on success
 */
export function findLegislatorsByLocation(
  requestParams: string[]
): LegislatorSearchResults {
  const legislatorSearchResults: LegislatorSearchResults = {
    status: Status.Loading,
    errorMessage: '',
    data: [],
  };
  const { loading, error, data } = useQuery(
    FIND_LEGISLATORS_BY_LOCATION_QUERY,
    {
      variables: {
        latitude: parseFloat(requestParams[0]),
        longitude: parseFloat(requestParams[1]),
      },
    }
  );

  if (loading) {
    return legislatorSearchResults;
  }
  if (error) {
    legislatorSearchResults.status = Status.Error;
    legislatorSearchResults.errorMessage = error.message;
    return legislatorSearchResults;
  }
  legislatorSearchResults.status = Status.Success;
  let wrapper: {
    node: {
      id: '';
      name: '';
    };
  };
  for (wrapper of data.people.edges) {
    legislatorSearchResults.data.push({
      id: wrapper.node.id,
      name: wrapper.node.name,
    });
  }
  return legislatorSearchResults;
}
