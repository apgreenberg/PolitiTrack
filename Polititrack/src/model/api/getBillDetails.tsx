import { gql, useQuery } from '@apollo/client';
import { Status } from './helpers';
import axios, { AxiosResponse } from 'axios';
import OPENSTATES_OAUTH_TOKEN from '../../types/env';

export interface BillDetails {
  status: Status;
  errorMessage: string;
  data: BillDetailNode;
}

export interface BillDetailNode {
  title: string;
  from: string;
  openstatesUrl: string;
  updatedAt: string;
}

export interface GetBillDetailsResponse {
  data: {
    bill: {
      fromOrganization: {
        name: string;
      };
      title: string;
      openstatesUrl: string;
      updatedAt: string;
    };
  };
}

export const BILL_DETAILS_QUERY = `
  query getBillDetails($id: String!) {
    bill(id: $id) {
      title
      openstatesUrl
      updatedAt
      fromOrganization {
        name
      }
    }
  }
`;

/**
 * Gets the bill details
 * @param requestParams - the request parameters (bill id)
 * @returns - the status of the request (loading, error or success) and the bill details on success
 */
export function getBillDetails(requestParams: string[]): BillDetails {
  const billDetails: BillDetails = {
    status: Status.Loading,
    errorMessage: '',
    data: { title: '', from: '', openstatesUrl: '', updatedAt: '' },
  };
  const { loading, error, data } = useQuery(
    gql`
      ${BILL_DETAILS_QUERY}
    `,
    {
      variables: {
        id: requestParams[0],
      },
    }
  );

  if (loading) {
    return billDetails;
  }
  if (error) {
    billDetails.status = Status.Error;
    billDetails.errorMessage = error.message;
    return billDetails;
  }
  billDetails.status = Status.Success;
  billDetails.data.title = data.bill.title;
  billDetails.data.openstatesUrl = data.bill.openstatesUrl;
  billDetails.data.updatedAt = data.bill.updatedAt;
  billDetails.data.from = data.bill.fromOrganization.name;
  return billDetails;
}

/**
 * Gets the bill details but using Axios instead of Apollo hooks
 * Necessary for background updates
 * @param requestParams - the request parameters (bill id)
 * @returns - the status of the request (loading, error or success) and the bill details on success
 */
export async function getBillDetailsAxios(
  requestParams: string[]
): Promise<BillDetails> {
  const billDetails: BillDetails = {
    status: Status.Loading,
    errorMessage: '',
    data: { title: '', from: '', openstatesUrl: '', updatedAt: '' },
  };
  try {
    const response: AxiosResponse<GetBillDetailsResponse> = await axios({
      url: 'https://openstates.org/graphql',
      method: 'post',
      headers: {
        'X-API-KEY': `${OPENSTATES_OAUTH_TOKEN}`,
      },
      data: {
        query: BILL_DETAILS_QUERY,
        variables: {
          id: requestParams[0],
        },
      },
    });
    billDetails.status = Status.Success;
    billDetails.data.title = response.data.data.bill.title;
    billDetails.data.openstatesUrl = response.data.data.bill.openstatesUrl;
    billDetails.data.updatedAt = response.data.data.bill.updatedAt;
    billDetails.data.from = response.data.data.bill.fromOrganization.name;
    return billDetails;
  } catch {
    (error: Error) => {
      billDetails.status = Status.Error;
      billDetails.errorMessage = error.message;
      return billDetails;
    };
  }
  return billDetails;
}
