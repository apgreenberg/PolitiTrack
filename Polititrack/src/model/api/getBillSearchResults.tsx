import { gql, useQuery } from '@apollo/client';
import { Status } from './helpers';

export interface BillSearchResults {
  status: Status;
  errorMessage: string;
  jurisdiction: string;
  billType: string;
  data: BillSearchNode[];
}

export interface BillSearchNode {
  id: string;
  name: string;
}

export const BILL_SEARCH_QUERY = gql`
  query getBillSearch(
    $jurisdiction: String!
    $session: String!
    $searchQuery: String!
  ) {
    bills(
      jurisdiction: $jurisdiction
      session: $session
      searchQuery: $searchQuery
      first: 30
    ) {
      edges {
        node {
          title
          id
        }
      }
    }
  }
`;

/**
 * Gets the search result bills
 * @param requestParams - the request
 *        legislature - the legislature/jurisdiction to search
 *        session - the legislative session to search
 *        billType - the bill subject that the user wants to lookup
 * @returns - a list of bills meeting the search criteria
 */
export function getBillSearchResults(
  requestParams: string[]
): BillSearchResults {
  const billSearchResults: BillSearchResults = {
    status: Status.Loading,
    errorMessage: '',
    jurisdiction: requestParams[0],
    billType: requestParams[2],
    data: [],
  };
  const { loading, error, data } = useQuery(BILL_SEARCH_QUERY, {
    variables: {
      jurisdiction: requestParams[0],
      session: requestParams[1],
      searchQuery: requestParams[2],
    },
  });

  if (loading) {
    return billSearchResults;
  }
  if (error) {
    billSearchResults.status = Status.Error;
    billSearchResults.errorMessage = error.message;
    return billSearchResults;
  }
  billSearchResults.status = Status.Success;
  let wrapper: {
    node: {
      id: '';
      title: '';
    };
  };
  for (wrapper of data.bills.edges) {
    billSearchResults.data.push({
      id: wrapper.node.id,
      name: wrapper.node.title,
    });
  }
  return billSearchResults;
}
