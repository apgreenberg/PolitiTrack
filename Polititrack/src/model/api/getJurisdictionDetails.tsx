import { gql, useQuery } from '@apollo/client';
import { Status } from './helpers';
import axios, { AxiosResponse } from 'axios';
import OPENSTATES_OAUTH_TOKEN from '../../types/env';

export interface JurisdictionDetails {
  status: Status;
  errorMessage: string;
  data: JurisdictionDetailNode;
}

export interface JurisdictionDetailNode {
  identifier: string;
  name: string;
  inSession: boolean;
  startDate: string;
  endDate: string;
}

export interface GetJurisdictionDetailsResponse {
  data: {
    jurisdiction: {
      legislativeSessions: {
        edges: GetJurisdictionDetailsResponseNode[];
      };
    };
  };
}

export interface GetJurisdictionDetailsResponseNode {
  node: {
    identifier: string;
    name: string;
    startDate: string;
    endDate: string;
  };
}

export const JURISDICTION_DETAILS_QUERY = `
  query getJurisdictionDetails($id: String!) {
    jurisdiction(id: $id) {
      legislativeSessions(first: 30) {
        edges {
          node {
            identifier
            name
            startDate
            endDate
          }
        }
      }
    }
  }
`;

/**
 * Gets the jurisdiction details
 * @param requestParams - the request parameters (jurisdiction id)
 * @returns - the status of the request (loading, error or success) and the jurisdiction details on success
 */
export function getJurisdictionDetails(
  requestParams: string[]
): JurisdictionDetails {
  const { loading, error, data } = useQuery(
    gql`
      ${JURISDICTION_DETAILS_QUERY}
    `,
    {
      variables: { id: requestParams[0] },
    }
  );

  const jurisdictionDetails: JurisdictionDetails = {
    status: Status.Loading,
    errorMessage: '',
    data: {
      identifier: '',
      name: 'Currently not in session',
      inSession: false,
      startDate: 'NA',
      endDate: 'NA',
    },
  };

  if (loading) {
    return jurisdictionDetails;
  }
  if (error) {
    jurisdictionDetails.status = Status.Error;
    jurisdictionDetails.errorMessage = error.message;
    return jurisdictionDetails;
  }
  jurisdictionDetails.status = Status.Success;
  findCurrentSession(
    jurisdictionDetails,
    data.jurisdiction.legislativeSessions.edges
  );
  return jurisdictionDetails;
}

/**
 * Same as getJurisdictionDetails, but uses Axios instead of Apollo hooks
 * Necessary for background updates
 * @param requestParams - the request parameters (jurisdiction id)
 * @returns - the status of the request (error or success) and the jurisdiction details on success
 */
export async function getJurisdictionDetailsAxios(
  requestParams: string[]
): Promise<JurisdictionDetails> {
  const jurisdictionDetails: JurisdictionDetails = {
    status: Status.Loading,
    errorMessage: '',
    data: {
      identifier: '',
      name: 'Currently not in session',
      inSession: false,
      startDate: 'NA',
      endDate: 'NA',
    },
  };
  try {
    const response: AxiosResponse<GetJurisdictionDetailsResponse> = await axios(
      {
        url: 'https://openstates.org/graphql',
        method: 'post',
        headers: {
          'X-API-KEY': `${OPENSTATES_OAUTH_TOKEN}`,
        },
        data: {
          query: JURISDICTION_DETAILS_QUERY,
          variables: {
            id: requestParams[0],
          },
        },
      }
    );
    jurisdictionDetails.status = Status.Success;
    findCurrentSession(
      jurisdictionDetails,
      response.data.data.jurisdiction.legislativeSessions.edges
    );
    return jurisdictionDetails;
  } catch {
    (error: Error) => {
      jurisdictionDetails.status = Status.Error;
      jurisdictionDetails.errorMessage = error.message;
      return jurisdictionDetails;
    };
  }
  return jurisdictionDetails;
}

/**
 * Given an array of legislative sessions, finds the current one if possible
 * @param jurisdictionDetails - the current jurisidiction details
 * @param sessions - the possible legislative sessions
 */
function findCurrentSession(
  jurisdictionDetails: JurisdictionDetails,
  sessions: GetJurisdictionDetailsResponseNode[]
): void {
  let wrapper = {
    node: {
      identifier: '',
      name: '',
      startDate: '',
      endDate: '',
    },
  };
  for (wrapper of sessions) {
    if (Date.parse(wrapper.node.endDate) > Date.parse(Date())) {
      jurisdictionDetails.data.identifier = wrapper.node.identifier;
      jurisdictionDetails.data.name = wrapper.node.name;
      jurisdictionDetails.data.inSession = true;
      jurisdictionDetails.data.startDate = wrapper.node.startDate;
      jurisdictionDetails.data.endDate = wrapper.node.endDate;
      break;
    }
  }
}
