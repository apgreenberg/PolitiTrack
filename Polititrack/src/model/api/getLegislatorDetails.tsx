import { gql, useQuery } from '@apollo/client';
import { Status } from './helpers';

export interface LegislatorDetails {
  status: Status;
  errorMessage: string;
  data: LegislatorDetailNode;
}

export interface LegislatorDetailNode {
  email: string;
  image: string;
  party: string;
}

export const LEGISLATOR_DETAILS_QUERY = gql`
  query getLegislatorDetails($id: ID!) {
    person(id: $id) {
      email
      image
      primaryParty
    }
  }
`;

/**
 * Gets the legislator details
 * @param requestParams - the request parameters (legislator id)
 * @returns - the status of the request (loading, error or success) and the legislator details on success
 */
export function getLegislatorDetails(
  requestParams: string[]
): LegislatorDetails {
  const legislatorDetails: LegislatorDetails = {
    status: Status.Loading,
    errorMessage: '',
    data: { email: '', image: '', party: '' },
  };
  const { loading, error, data } = useQuery(LEGISLATOR_DETAILS_QUERY, {
    variables: {
      id: requestParams[0],
    },
  });

  if (loading) {
    return legislatorDetails;
  }
  if (error) {
    legislatorDetails.status = Status.Error;
    legislatorDetails.errorMessage = error.message;
    return legislatorDetails;
  }
  legislatorDetails.status = Status.Success;
  legislatorDetails.data.email = data.person.email;
  legislatorDetails.data.image = data.person.image;
  legislatorDetails.data.party = data.person.primaryParty;
  return legislatorDetails;
}
