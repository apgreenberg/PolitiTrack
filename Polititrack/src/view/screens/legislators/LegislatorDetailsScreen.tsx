import React from 'react';
import { StyleSheet, Image, Text, View, SafeAreaView } from 'react-native';
import 'react-native-gesture-handler';
import * as Linking from 'expo-linking';
import { StackScreenProps } from '@react-navigation/stack';

import { StackParams } from '../../../navigation/StackParams';
import { getLegislatorDetails } from '../../../model/api/getLegislatorDetails';

import CustomButton from '../../components/CustomButton';
import LoadingScreen from '../LoadingScreen';
import ErrorScreen from '../ErrorScreen';

import text from '../../styles/text';
import { Status } from '../../../model/api/helpers';
import { updateLastUpdated } from '../../../model/database/handleContent';

type Props = StackScreenProps<StackParams, 'Legislator Details'>;

/**
 * Displayes the legislator details for a given state
 * @param - the legislator name and id to request the details for
 * @returns - the functional component representing the screen
 */
const LegislatorDetailsScreen: React.FC<Props> = ({ route, navigation }) => {
  React.useEffect(() => {
    (async () => updateLastUpdated(route.params.content))();
  }, []);
  const legislatorDetails = getLegislatorDetails([route.params.content.id]);

  if (legislatorDetails.status === Status.Loading) {
    return <LoadingScreen />;
  }
  if (legislatorDetails.status === Status.Error) {
    return <ErrorScreen message={legislatorDetails.errorMessage} />;
  }

  return (
    <SafeAreaView>
      <Text style={text.largeText}>{route.params.content.name}</Text>
      <View style={{ borderBottomWidth: 2, padding: 5, flexDirection: 'row' }}>
        <Image
          style={{ width: 70, height: 70 }}
          source={{
            uri: legislatorDetails.data.image,
          }}
        />
        <View style={{ justifyContent: 'center', flex: 1 }}>
          <Text style={text.mediumText}>
            Party: {legislatorDetails.data.party}
          </Text>
          <Text
            style={[text.mediumText, { color: 'blue' }]}
            onPress={() => {
              Linking.openURL('mailto:' + legislatorDetails.data.email);
            }}
          >
            {legislatorDetails.data.email}
          </Text>
        </View>
      </View>
      <View style={styles.buttonPanelContainer}>
        <View style={styles.buttonContainer}>
          <CustomButton
            type="Committees"
            destination="Legislator Committees"
            params={route.params.content}
          ></CustomButton>
        </View>
        <View style={styles.buttonContainer}>
          <CustomButton
            type="Voting History"
            destination="Legislator Voting History"
            params={route.params.content}
          ></CustomButton>
        </View>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    alignItems: 'center',
    height: '50%',
    width: '100%',
  },
  buttonPanelContainer: {
    padding: 20,
    justifyContent: 'space-around',
    height: '70%',
  },
});

export default LegislatorDetailsScreen;
