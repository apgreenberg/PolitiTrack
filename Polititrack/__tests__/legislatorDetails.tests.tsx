import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import { GraphQLError } from 'graphql';

import { wait, legislatorProps } from './common';
import LegislatorDetailsScreen from '../src/view/screens/legislators/LegislatorDetailsScreen';
import { LEGISLATOR_DETAILS_QUERY } from '../src/model/api/getLegislatorDetails';

/**
 * Tests the Legislator Details Screen
 */
describe('Legislator Detail tests', () => {
  test('legislator loading and success screen', async () => {
    const legislatorDetailsMockSuccess = {
      request: {
        query: LEGISLATOR_DETAILS_QUERY,
        variables: {
          id: 'ocd-person/bc409660-f360-4624-914f-ba785bbf9b7b',
        },
      },
      result: {
        data: {
          person: {
            email: 'Assistance@StateRepCarolAmmons.com',
            image:
              'http://ilga.gov/images/members/{BD4B7A5C-6D24-4E82-9C18-D5A0F3C42F6C}.jpg',
            primaryParty: 'Democratic',
          },
        },
      },
    };
    const component = TestRenderer.create(
      <MockedProvider
        mocks={[legislatorDetailsMockSuccess]}
        addTypename={false}
      >
        <LegislatorDetailsScreen {...legislatorProps} />
      </MockedProvider>
    );
    const loadingScreen = component.toJSON();
    expect(loadingScreen).toMatchSnapshot();

    await wait();

    const successScreen = component.toJSON();
    expect(successScreen).toMatchSnapshot();
  });
  test('Legislator details network error screen', async () => {
    const legislatorDetailsMockNetworkError = {
      request: {
        query: LEGISLATOR_DETAILS_QUERY,
        variables: {
          id: 'ocd-person/bc409660-f360-4624-914f-ba785bbf9b7b',
        },
      },
      error: new Error('Network error!'),
    };
    const component = TestRenderer.create(
      <MockedProvider
        mocks={[legislatorDetailsMockNetworkError]}
        addTypename={false}
      >
        <LegislatorDetailsScreen {...legislatorProps} />
      </MockedProvider>
    );
    await wait();

    const networkErrorScreen = component.toJSON();
    expect(networkErrorScreen).toMatchSnapshot();
  });
  test('Legislature details graphql error screen', async () => {
    const legislatorDetailsMockGraphQLError = {
      request: {
        query: LEGISLATOR_DETAILS_QUERY,
        variables: {
          id: 'ocd-person/bc409660-f360-4624-914f-ba785bbf9b7b',
        },
      },
      result: {
        errors: [new GraphQLError('GraphQL Error!')],
      },
    };
    const component = TestRenderer.create(
      <MockedProvider
        mocks={[legislatorDetailsMockGraphQLError]}
        addTypename={false}
      >
        <LegislatorDetailsScreen {...legislatorProps} />
      </MockedProvider>
    );
    await wait();

    const graphqlErrorScreen = component.toJSON();
    expect(graphqlErrorScreen).toMatchSnapshot();
  });
});
