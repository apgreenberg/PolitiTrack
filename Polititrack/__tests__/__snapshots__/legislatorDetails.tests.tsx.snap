// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legislator Detail tests Legislator details network error screen 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 12,
        "textAlign": "center",
      }
    }
  >
    Error: 
    Network error!
  </Text>
</View>
`;

exports[`Legislator Detail tests Legislature details graphql error screen 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 12,
        "textAlign": "center",
      }
    }
  >
    Error: 
    GraphQL Error!
  </Text>
</View>
`;

exports[`Legislator Detail tests legislator loading and success screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Loading
  </Text>
</View>
`;

exports[`Legislator Detail tests legislator loading and success screen 2`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "textAlign": "center",
      }
    }
  >
    Carol Ammons
  </Text>
  <View
    style={
      Object {
        "borderBottomWidth": 2,
        "flexDirection": "row",
        "padding": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://ilga.gov/images/members/{BD4B7A5C-6D24-4E82-9C18-D5A0F3C42F6C}.jpg",
        }
      }
      style={
        Object {
          "height": 70,
          "width": 70,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Party: 
        Democratic
      </Text>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "fontSize": 16,
              "textAlign": "center",
            },
            Object {
              "color": "blue",
            },
          ]
        }
      >
        Assistance@StateRepCarolAmmons.com
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "height": "70%",
        "justifyContent": "space-around",
        "padding": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "50%",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#dae8eb",
            "borderBottomWidth": 2,
            "borderColor": "#9ca7b8",
            "borderLeftWidth": 1,
            "borderRadius": 5,
            "borderRightWidth": 1,
            "borderTopWidth": 2,
            "height": "90%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "90%",
          }
        }
        testID="customButton"
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "textAlign": "center",
            }
          }
        >
          Committees
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "50%",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#dae8eb",
            "borderBottomWidth": 2,
            "borderColor": "#9ca7b8",
            "borderLeftWidth": 1,
            "borderRadius": 5,
            "borderRightWidth": 1,
            "borderTopWidth": 2,
            "height": "90%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "90%",
          }
        }
        testID="customButton"
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "textAlign": "center",
            }
          }
        >
          Voting History
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
